File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        left = 0
        right = len(nums)-1
        done = False

        nums.sort()

        while not done:
            if left > len(nums)-1 or right > len(nums)-1:
                done = True
            else:
                left_val = nums[left]
                right_val = nums[right]
                if val < left_val or val > right_val:
                        done = True
                elif left_val == val:
                    nums.pop(left)
                    right -= 1
                elif right_val == val:
                    nums.pop(right)
                    right -= 1
                else:
                    left += 1
                    right -= 1
                    
        
        return len(nums)

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1; 

        while(l <= r) {
            int m = l + (r - l) / 2;

            if(nums[m] == target) return m;

            if (nums[m] > nums[r]) {
                if (nums[m] > target
                    && target >= nums[l]) {
                        // to the left half
                        r = m-1;
                } else {
                        // to the right half
                        l = m+1;
                }

            } else {  // nums[m] < nums[r]
                if (nums[m] < target
                    && target <= nums[r]) {
                        // to the right half
                        l = m+1;
                    } else {
                        // to the left half
                        r = m-1;
                    }
            }
        }

        return -1;
    }
}
